grouper/repositories/permission_grant.py

```diff
@@ -172,35 +207,66 @@ def group_grants_for_permission(self, name, include_disabled_groups=False):
             for g in grants.all()
         ]
 
-    def service_account_grants_for_permission(self, name):
-        # type: (str) -> List[ServiceAccountPermissionGrant]
-        permission = Permission.get(self.session, name=name)
-        if not permission or not permission.enabled:
+    def permission_grants_for_group(self, name):
+        # type: (str) -> List[GroupPermissionGrant]
+        """Return all permission grants a group has.
+
+        TODO(rra): Currently does not expand permission aliases, and therefore doesn't match the
+        graph behavior.  Use with caution until that is fixed.
+        """
+        now = datetime.utcnow()
+        group = Group.get(self.session, name=name)
+        if not group or not group.enabled:
             return []
-        grants = (
+        group_ids = [group.id]
+
+        # Now, get the parent groups of that group and so forth until we run out of levels of the
```

Comment:
While it is awkward in the repository world, could we pull this logic out into its own helper? Having such a large and perfectly identical chunk of code be duplicated kinda hurts my soul :P

Reply:
Yeah, I realized that after I went home. Now fixed in a subsequent diff in this same PR.